modules/core/src/main/java/org/apache/ignite/internal/processors/service/ServiceContextImpl.java

```diff
@@ -56,8 +59,9 @@
     @GridToStringExclude
     private final ExecutorService exe;
 
-    /** Methods reflection cache. */
-    private final ConcurrentMap<GridServiceMethodReflectKey, Method> mtds = new ConcurrentHashMap<>();
+    /** Keeps cached data bound to service method. */
+    private final ConcurrentMap<GridServiceMethodReflectKey, IgniteBiTuple<Method, HistogramMetricImpl>> mtds =
```

Comment:
I suggest make code more readable and create a separate map for metrics. `ConcurrentMap<Method, HistogramMetricImpl>`

Reply:
The map has been split.